apps/scoutgame/components/layout/UserProvider.tsx

```
@@ -0,0 +1,35 @@
+'use client';
+
+import type { ReactNode } from 'react';
+import { createContext, useCallback, useContext, useMemo, useState } from 'react';
+
+import type { SessionUser } from 'lib/session/getUserFromSession';
+
+type UserContext = {
+  user: SessionUser | null;
+  updateUser: (updatedUser: Partial<SessionUser>) => void;
+};
+
+export const UserContext = createContext<Readonly<UserContext | null>>(null);
+
+export function UserProvider({ children, userSession }: { children: ReactNode; userSession: SessionUser | null }) {
+  const [user, setUser] = useState<SessionUser | null>(userSession);
+
+  const updateUser = useCallback((_updatedUser: Partial<SessionUser>) => {
```

You could throw an error here, incase someone tries to call it. "FIXME: User update is not implemented!"

---

Added!